appcli/backup_manager/remote_strategy.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# # -*- coding: utf-8 -*-
+
+"""
+Remote backup strategy classes.
+________________________________________________________________________________
+
+Created by brightSPARK Labs
+www.brightsparklabs.com
+"""
+
+# standard libraries
+import os
+import urllib.parse
+from pathlib import Path
+
+# vendor libraries
+import boto3
+from botocore.exceptions import ClientError
+
+# local libraries
+from appcli.crypto.cipher import Cipher
+from appcli.logger import logger
+
+
+class RemoteStrategy:
+    """
+    Base class for a remote backup strategy.
+    """
+
+    # --------------------------------------------------------------------------
+    # CONSTRUCTOR
+    # --------------------------------------------------------------------------
+    def __init__(self, conf, key_file: Path):
+        self.name = conf["name"]
+        self.type = conf["type"]
+        self.frequency = conf["frequency"]
+        self.configuration = conf["configuration"]
+        self.key_file = key_file
+
+    # ------------------------------------------------------------------------------
+    # PUBLIC METHODS
+    # ------------------------------------------------------------------------------
+    def backup(self, backup_filename):
+        pass
+
+
+class AwsS3Strategy(RemoteStrategy):
+    """
+    Remote backup strategy for pushing a backup to an S3 bucket.
+    Implements RemoteStrategy.
+    """
+
+    # --------------------------------------------------------------------------
+    # CONSTRUCTOR
+    # --------------------------------------------------------------------------
+    def __init__(self, conf, key_file: Path):
+        super().__init__(conf, key_file)
+
+        cipher = Cipher(self.key_file)
+
+        self.s3_bucket = self.configuration["bucket_name"]
+        # Both the access_key and secret_key should be encrypted, decrypt them.
+        self.s3_access_key = cipher.decrypt(self.configuration["access_key"])
```

Comment:
Leave access key unencryptetd. This makes it easier to revoke the key if needed so it can be clearly seen in the config.